compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -180,8 +180,9 @@ trait QuotesAndSplices {
     else
       val tree1 = typedSelect(untpd.Select(tree.expr, tpnme.Underlying), pt)(using spliceContext).withSpan(tree.span)
       val msg = em"Consider using canonical type reference ${tree1.tpe} instead"
-      if sourceVersion.isAtLeast(`3.1-migration`) then report.error(msg, tree.srcPos)
-      else report.warning(msg, tree.srcPos)
+      // if sourceVersion.isAtLeast(`3.1-migration`) then
+      report.error(msg, tree.srcPos)
+      // else report.warning(msg, tree.srcPos)
```

Comment:
So we decide to enforce it in M1? That's fine, just make sure it's not an accidental change.

Reply:
I need to revert this change

Reply:
Reverted